lib/ansible/modules/cloud/digital_ocean/digital_ocean_image_facts.py

```diff
@@ -0,0 +1,170 @@
+options:
+  image_type:
+    description:
+     - Specifies the type of image facts to be retrived.
+     - If set to C(application), then facts are gathered related to all application images.
+     - If set to C(distribution), then facts are gathered related to all distribution images.
+     - If set to C(private), then facts are gathered related to all private images.
+     - If not set to any of above, then facts are gathered related to all images.
+    default: 'all'
+    choices: [ 'all', 'application', 'distribution', 'private' ]
+    required: false
+  oauth_token:
+    description:
+     - DigitalOcean OAuth token.
+    required: true
+    aliases: ['DO_API_TOKEN', 'DO_API_KEY', 'DO_OAUTH_TOKEN']
+
+notes:
+  - Two environment variables can be used, DO_API_KEY, DO_OAUTH_TOKEN and DO_API_TOKEN.
+    They both refer to the v2 token.
+
+requirements:
+  - "python >= 2.6"
+'''
+
+
+EXAMPLES = '''
+- name: Gather facts about all images
+  digital_ocean_image_facts:
+    image_type: all
+    oauth_token: "{{ oauth_token }}"
+
+- name: Gather facts about application images
+  digital_ocean_image_facts:
+    image_type: application
+    oauth_token: "{{ oauth_token }}"
+
+- name: Gather facts about distribution images
+  digital_ocean_image_facts:
+    image_type: distribution
+    oauth_token: "{{ oauth_token }}"
+
+'''
+
+
+RETURN = '''
+data:
+    description: DigitalOcean image facts
+    returned: success
+    type: list
+    sample: [
+        {
+            "created_at": "2018-02-02T07:11:43Z",
+            "distribution": "CoreOS",
+            "id": 31434061,
+            "min_disk_size": 20,
+            "name": "1662.1.0 (beta)",
+            "public": true,
+            "regions": [
+                "nyc1",
+                "sfo1",
+                "nyc2",
+                "ams2",
+                "sgp1",
+                "lon1",
+                "nyc3",
+                "ams3",
+                "fra1",
+                "tor1",
+                "sfo2",
+                "blr1"
+            ],
+            "size_gigabytes": 0.42,
+            "slug": "coreos-beta",
+            "type": "snapshot"
+        },
+    ]
+'''
+
+from traceback import format_exc
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.digital_ocean import DigitalOceanHelper
+from ansible.module_utils._text import to_native
+
+
+def core(module):
+    image_type = module.params['image_type']
+
+    rest = DigitalOceanHelper(module)
+
+    # Check if api_token is valid or not
+    response = rest.get('account')
+    if response.status_code == 401:
+        module.fail_json(msg='Failed to login using api_token, please verify '
+                             'validity of api_token')
+    base_url = 'images?'
+    if image_type == 'distribution':
+        base_url += "type=distribution&"
+    elif image_type == 'application':
+        base_url += "type=application&"
+    elif image_type == 'private':
+        base_url += "private=true&"
```

Comment:
I'm quiet confused about this, the distribution and **application** images can't be private? They also have a param **public** param for that types.
Shouldn't **private**/**public** be a different parameter?

Reply:
I am also not sure. API documentation does not state anything about it [here](https://developers.digitalocean.com/documentation/v2/#list-a-user-s-images)

Reply:
I believe this relates to team accounts. Users are able to create an image for their account and not expose it to additional team members.